sdk/hybridnetwork/Azure.ResourceManager.HybridNetwork/src/Generated/Models/AzureStorageAccountCredential.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.HybridNetwork.Models
+{
+    /// <summary> The azure storage account credential definition. </summary>
+    public partial class AzureStorageAccountCredential : ArtifactAccessCredential
+    {
+        /// <summary> Initializes a new instance of AzureStorageAccountCredential. </summary>
+        internal AzureStorageAccountCredential()
+        {
+            ContainerCredentials = new ChangeTrackingList<AzureStorageAccountContainerCredential>();
+            CredentialType = CredentialType.AzureStorageAccountToken;
+        }
+
+        /// <summary> Initializes a new instance of AzureStorageAccountCredential. </summary>
+        /// <param name="credentialType"> The credential type. </param>
+        /// <param name="storageAccountId"> The storage account Id. </param>
+        /// <param name="containerCredentials"> The containers that could be accessed using the current credential. </param>
+        /// <param name="expiry"> The UTC time when credential will expire. </param>
+        internal AzureStorageAccountCredential(CredentialType credentialType, string storageAccountId, IReadOnlyList<AzureStorageAccountContainerCredential> containerCredentials, DateTimeOffset? expiry) : base(credentialType)
+        {
+            StorageAccountId = storageAccountId;
+            ContainerCredentials = containerCredentials;
+            Expiry = expiry;
+            CredentialType = credentialType;
+        }
+
+        /// <summary> The storage account Id. </summary>
+        public string StorageAccountId { get; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).